.github/workflows/main.yml

```diff
@@ -0,0 +1,19 @@
+on:
+  push:
+    branches: [ main ]
```

Comment:
currently we are only running this workflow when we push to `main` branch.

If we want to get fancy, we could have an additional run when we make a release, and tag the docker image with the release version (currently we are always uploading w/ "latest" tag.